File: leetcode/0119-pascals-triangle-ii/solution.py
from typing import List


class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        row = [1] * (rowIndex + 1)
        for j in range(1, rowIndex + 1):
            row[j] = row[j - 1] * (rowIndex - j + 1) // j
        return row

File: leetcode/0119-pascals-triangle-ii/solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<long long> row(rowIndex + 1, 1);
        for (int j = 1; j <= rowIndex; j++) {
            row[j] = row[j - 1] * (rowIndex - j + 1) / j;
        }
        return vector<int>(row.begin(), row.end());
    }
};
